transcom/mymove
# MB-6752 Admin UI: Add userId field to Admin & Office Users pages

## Description

This PR adds the `User Id` field to the Admin & Office Users list, detail, and edit pages. It also does some extra clean-up work around this part of the app:

* Adds full cypress tests for all pages for Admin & Office Users.
* Moves the `AdminUsers` and `OfficeUsers` folders from the deprecated `src/scenes/SystemAdmin` directory to the new standard `src/pages/Admin` directory.
* Adds an `.eslintrc` file to the `src/pages/Admin` directory that ignores the `react/jsx-no-props-spreading` rule for all files in the directory. We have this rule ignored in literally every page for the Admin UI, and we really should just ignore it for the whole Admin folder.
* Cleans up some miscellaneous bugs in the admin interface, such as removing the checkboxes that do nothing in the moves list, fixing a typo on the `middleInitials` field for office users, and updating some imports.

## Setup

Run `make e2e_test` to run the cypress tests (`cypress/integration/admin/adminUsers.js` and `officeUsers.js` are the new ones).

To test the interface, run:

```sh
make server_run
make admin_client_run
```

Select "Local Sign-in." Verify that you can see the "User Id" field populated on these pages:
* Office Users list page (the home screen)
* Office Users detail page
* Office Users edit page (as a disabled input)
* Admin Users list page
* Admin Users detail page
* Admin Users edit page (as a disabled input)

Try grabbing the user ID value from a record or two and search for it in the Users list page. You should find the record you want! 

## Code Review Verification Steps


## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6752) for this change.

## Screenshots

Adding the "User Id" field to the Admin & Office Users list, detail, and edit pages:
![Screen Shot 2021-03-03 at 11 28 17 AM](https://user-images.githubusercontent.com/62574202/109847316-aab34e80-7c14-11eb-93c2-78cab9fbe247.png)
![Screen Shot 2021-03-03 at 11 28 27 AM](https://user-images.githubusercontent.com/62574202/109847328-aedf6c00-7c14-11eb-9436-279f312407f1.png)
![Screen Shot 2021-03-03 at 11 28 56 AM](https://user-images.githubusercontent.com/62574202/109846533-e7cb1100-7c13-11eb-863a-366e9b09f538.png)
![Screen Shot 2021-03-03 at 11 29 13 AM](https://user-images.githubusercontent.com/62574202/109846552-ec8fc500-7c13-11eb-91c1-615b6cfcad26.png)

Removing the superfluous checkboxes from the Moves list page:
![Screen Shot 2021-03-03 at 11 29 35 AM](https://user-images.githubusercontent.com/62574202/109846722-15b05580-7c14-11eb-9894-6c917ed5e3db.png)
